lib/parse/ClassLoader.ts

```diff
@@ -457,11 +457,17 @@ export class ClassLoader {
   /**
    * Load a class, and get all class elements from it.
    * @param packageName Package name we are importing from.
-   * @param fileName A file path.
+   * @param filePath A file path.
+   * @returns {Promise<ClassElements & { resolvedPath: string }>} Promise of the class elements along with
+   * the resolved file path that was used to load these class elements.
    */
-  public async loadClassElements(packageName: string, fileName: string): Promise<ClassElements> {
-    const ast = await this.resolutionContext.parseTypescriptFile(fileName);
-    return this.getClassElements(packageName, fileName, ast);
+  public async loadClassElements(packageName: string, filePath: string): Promise<
+  ClassElements & { resolvedPath: string }
+  > {
+    const resolvedPath = await this.resolutionContext.resolveTypesPath(filePath);
```

Comment:
Path resolution has been added here instead of ClassFinder.ts as it seemed the most elegant solution for now. The idea is that `loadClassElements` on a filePath gives you the class elements and the path used to obtain them